File: 1200. Minimum Absolute Difference.cpp
class Solution {
public:
    std::vector<std::vector<int>> minimumAbsDifference(std::vector<int>& arr) {
        std::vector<std::vector<int>> ans;
        std::sort(arr.begin(), arr.end());
        int minDiff = INT_MAX;
        for (size_t i = 1; i < arr.size(); i++) {
            minDiff = std::min(minDiff, arr[i] - arr[i - 1]);
        }

        for (size_t i = 1; i < arr.size(); i++) {
            if (minDiff == arr[i] - arr[i - 1]) {
                ans.push_back({arr[i - 1], arr[i]});
            }
        }

        return ans;
    }
};

File: 1200. Minimum Absolute Difference.java
class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] arr) {
        List<List<Integer>> ans = new ArrayList<>();
        Arrays.sort(arr);
        int minDiff = Integer.MAX_VALUE;
        for(int i = 1; i < arr.length; i++) minDiff = Math.min(minDiff, arr[i] - arr[i - 1]);

        for(int i = 1; i < arr.length; i++){
            if(minDiff == arr[i] - arr[i - 1]){
                List<Integer> temp = new ArrayList<>();
                temp.add(arr[i - 1]);
                temp.add(arr[i]);
                ans.add(temp);
            }
        }

        return ans;
    }
}